### ISSUE TITLE
brain example?

### ISSUE THREAD

**Issue by ntustison** [ha-ha-ha-han/UKBiobank_deep_pretrain#1]
Hey, this is great.  Thanks for making it available. 

Can you provide an actual brain image example in addition to the random noise example?

**Comment by ha-ha-ha-han** [comment]
> Hey, this is great. Thanks for making it available.
> 
> Can you provide an actual brain image example in addition to the random noise example?

Unfortunately, people can't put UK Biobank raw data on the web. Any suggestion for an openly available dataset without the need for additional preprocessing?

**Comment by ntustison** [comment]
Thanks.  Much appreciated and I certainly understand not sharing UK Biobank raw data.  

I actually forked your repository and added a human example python notebook which demonstrates the difficulties I'm having with your model/weights on actual human data.  Here's the [forked repository](https://github.com/ntustison/UKBiobank_deep_pretrain) and this is the [subdirectory](https://github.com/ntustison/UKBiobank_deep_pretrain/tree/master/human_brain_image_data_preprocessed) where I put two preprocessed human brain images.  They were skull stripped, bias corrected, and affinely warped to MNI152 as specified in your paper.   This is the [modified python notebook](https://github.com/ntustison/UKBiobank_deep_pretrain/blob/master/examples_human.ipynb) which uses those two human brain images for prediction but, as you can see, I'm not getting the expected accuracy.  My guess is that I might not be standardizing the intensities after loading, specifically this line here:

```
data = data/data.mean()
```

but I can't find those details in your paper.    Any help would be much appreciated.  Thanks.

**Comment by ha-ha-ha-han** [comment]
> Thanks. Much appreciated and I certainly understand not sharing UK Biobank raw data.
> 
> I actually forked your repository and added a human example python notebook which demonstrates the difficulties I'm having with your model/weights on actual human data. Here's the [forked repository](https://github.com/ntustison/UKBiobank_deep_pretrain) and this is the [subdirectory](https://github.com/ntustison/UKBiobank_deep_pretrain/tree/master/human_brain_image_data_preprocessed) where I put two preprocessed human brain images. They were skull stripped, bias corrected, and affinely warped to MNI152 as specified in your paper. 

Thanks a lot for adding human samples! 
Yes, this looks all good.

>This is the [modified python notebook](https://github.com/ntustison/UKBiobank_deep_pretrain/blob/master/examples_human.ipynb) which uses those two human brain images for prediction but, as you can see, I'm not getting the expected accuracy. My guess is that I might not be standardizing the intensities after loading, specifically this line here:
> 
> ```
> data = data/data.mean()
> ```
> 
> but I can't find those details in your paper. Any help would be much appreciated. Thanks.

Looks like the code is working. 
```
# Preprocessing
data = data/data.mean()
data = dpu.crop_center(data, (160, 192, 160))
```
This is correct and is what we used. 

May I ask what is the data source? Since the model is trained only with UK Biobank data, so that it may need transfer-learning for your protocol/scanner.

**Comment by ntustison** [comment]
Thanks Han.  I appreciate the feedback.  These two images were from my institution (University of Virginia).  I don't know what the parameters were but I can look.  

I actually have several publicly available data from different sites.  I'm going to run them through this pipeline and see if I can get better numbers.  I might need to come back and ask for your thoughts on the results but thanks for your help so far.

**Comment by ha-ha-ha-han** [comment]
No problem. We transferred the model to the PAC2019 challenge data (multisite, different age range than UKBiobank, 16-90yrs) and it saved some training time if you use UK Biobank-pretrained weights as an initialization. Maybe this can be helpful

**Comment by ballarinit** [comment]
Hi, thank you for the impressive work! I am currently testing the model on different data and the code is working well.
Any chance you could share more details about the MRI preprocessing or the code used for it? It would be great to have exactly the same steps here.

**Comment by ha-ha-ha-han** [comment]
> Hi, thank you for the impressive work! I am currently testing the model on different data and the code is working well.
> Any chance you could share more details about the MRI preprocessing or the code used for it? It would be great to have exactly the same steps here.

Thanks a lot for your interest in our work! The full UK Biobank preprocessing pipeline can be found [here]((https://git.fmrib.ox.ac.uk/falmagro/UK_biobank_pipeline_v_1)).

PS: pls consider star our GitHub repository, as we will keep updating the newly trained models and weights :)

**Comment by ha-ha-ha-han** [comment]
> Hi, thank you for the impressive work! I am currently testing the model on different data and the code is working well.
> Any chance you could share more details about the MRI preprocessing or the code used for it? It would be great to have exactly the same steps here.

Hi, I have just updated the readme page. Check it out :)
Btw, if you fine-tune the model in your own data, then it is not necessary that the preprocessing to be exactly the same.

PS: pls consider star and/or watch this repository if you find it helpful, as we will keep updating this repository for pretrained models and weights
